pkg/scalers/scaler.go

```diff
@@ -168,47 +168,106 @@ func GenerateMetricInMili(metricName string, value float64) external_metrics.Ext
 	}
 }
 
+type Option func(*configOptions)
+
+type configOptions struct {
+	parameter         string
+	useMetadata       bool
+	useAuthentication bool
+	useResolvedEnv    bool
+	isOptional        bool
+	defaultVal        interface{}
+	targetType        reflect.Type
+}
+
+func WithParameter(parameter string) Option {
```

Comment:
imho, parameter shouldn't be an option. Calling `getParameterFromConfigV2` without a parameter name should not be possible.